src/Microsoft.AspNet.Identity/SignInManager.cs

```diff
@@ -41,7 +42,7 @@ public class SignInManager<TUser> where TUser : class
             {
                 throw new ArgumentNullException(nameof(userManager));
             }
-            if (contextAccessor == null || contextAccessor.HttpContext == null)
```

Comment:
Yeah I think we just need to defer the Context guard into a helper method which will throw then if HttpContext is null, https://github.com/aspnet/Identity/issues/674 basically asked us not to get it in the ctor.